lib/index.js

```diff
@@ -262,6 +262,23 @@ class Study extends EventTarget {
         die();
       }).bind(this)
     )
+    this.on(
+      "uninstall-custom-reason", (function (reason) {
+        prefSvc.get('shield.debug') && console.log("uninstall-custom-reason", this.states);
+        this.states.push("uninstall-custom-reason");
+        if (this.flags.dying) {
+          this.final();
+          return;
+        }
+        this.flags.dying = true;
+        this.report(merge({}, telemetrySubset(this.config), {study_state: reason}), "shield");
+        let that = this;
```

Comment:
Don't need to set `that = this;` here, right? The [`=>` function will automatically bind `this` for `showSurvey`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#No_binding_of_this)?